docs/sdk/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,589 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! As our runtime should _never_ panic; this includes carefully handling [`Result`]/[`Option`]
+//! types, eliminating the possibility of integer overflows, converting between number types, or
+//! even replacing floating point usage with fixed point arithmetic to mitigate issues that come
+//! with floating point calculations.
+//!
+//! Intentional and predictable design should be our first and foremost
+//! property for ensuring a well running, safely designed system.
+//!
+//! ## Defensive Programming
+//!
+//! [Defensive programming](https://en.wikipedia.org/wiki/Defensive_programming) is a design paradigm that enables a particular program to continue
+//! running despite unexpected behavior. These unforseen circumstances may
+//! cause the program to stop or, in the Rust context, `panic!`. Defensive practices allow for
+//! these circumstances to be accounted for ahead of time and for them to be handled in a more
+//! graceful manner.
+//!
+//! The Polkadot SDK is both built to reflect these principles and to be used accordingly.
+//!
+//! ## General Practices
+//!
+//! When developing within the context of the a runtime, there is one golden rule:
+//!
+//! ***DO NOT PANIC***. There are some exceptions, such as critical operations being actually more
+//! dangerous than allowing the runtime to continue functioning (block authoring, consensus, etc).
+//!
+//! > It's important to make the differentiation between the **runtime** and **node**.  The runtime
+//! > refers to the core business logic of a Substrate-based chain, whereas the node refers to the
+//! > outer client which deals with telemetry and gossip from other nodes. For more information,
+//! > read about Substrate's architecture.
+//!
+//!  General guidelines:
+//!
+//! - Avoid any writing functions that could explicitly panic.  Directly using `unwrap()` for a
+//!   [`Result`] or common errors such as accessing an out of bounds index on a collection (using
+//!   safer methods to access collection types, i.e., `get()`) are both examples of what to
```

Comment:
I'd move the recommendation inside parenthesis to a phrase after listing both examples.